### DISCUSSION TITLE
Tech references to understand linkerd internals

### DISCUSSION THREAD

**Discussion by mrveera** [linkerd/linkerd2#5884] [Q&A] [answered]
Anyone here have reference to understand internals of linkerd2 other than code, We are having too many 503s
When we make call from service A to B, we know that call reaches outbound proxy of service A but it doesn't go to service B inbound proxy
I am new to Rust so looking for tech docs if we have any
or any debugging tricks ?

**Comment by mateiidavid** [comment, marked as answer]
Hey @veera83372 

I'll link below some of my favourite pieces written/shot by the maintaining team :) 

* [For 502s](https://linkerd.io/2.10/tasks/debugging-502s/)
* [Inside linkerd2-proxy](https://linkerd.io/2020/07/23/under-the-hood-of-linkerds-state-of-the-art-rust-proxy-linkerd2-proxy/)
* [Using the debugging sidecar](https://linkerd.io/2.10/tasks/using-the-debug-container/)
* [Deep dive (kubecon pres)](https://www.youtube.com/watch?v=NqjRqe0J98U)
* [Nondestructive forensics - debugging k8s services (kubecon pres on how to use linkerd to debug)](https://www.youtube.com/watch?v=fMtJFtKeENA&t=2s)

Think if you have a look at these you'll get a better feel for how to debug everything. There are a few tools e.g `tap` that can make your life a bit easier.

Hope this helps : )
